python/lsst/ap/association/packageAlerts.py

```diff
@@ -165,6 +177,80 @@ def createDiaSourceBBox(self, bboxSize):
             bbox = geom.Extent2I(bboxSize, bboxSize)
         return bbox
 
+    def createCcdDataCutout(self, cutout, sphPoint, photoCalib):
+        """Convert a cutout into a calibrate CCDData image.
+
+        Parameters
+        ----------
+        cutout : `lsst.afw.image.ExposureF`
+            Cutout to convert.
+        sphPoint : `lsst.geom.SpherePoint`
+            Center point of DiaSource on the sky.
+        photoCalib : `lsst.afw.image.PhotoCalib`
+            Calibrate object of the image the cutout is cut from.
+
+        Returns
+        -------
+        ccdData : `astropy.nddata.CCDData`
+            CCDData object storing the calibrate information from the input
+            difference or template image.
+        """
+        cutOutMinX = cutout.getBBox().minX - 1
+        cutOutMinY = cutout.getBBox().minY - 1
+        center = cutout.getWcs().skyToPixel(sphPoint)
+        calibCutout = photoCalib.calibrateImage(cutout.getMaskedImage())
+
+        cutoutWcs = wcs.WCS(naxis=2)
+        cutoutWcs.array_shape = (cutout.getBBox().getWidth(),
+                                 cutout.getBBox().getWidth())
+        cutoutWcs.wcs.crpix = [center.x - cutOutMinX, center.y - cutOutMinY]
```

Comment:
Why is the crpix at the corner rather than the center of the cutout?

Reply:
So, CCDData doesn't store absolute pixel locations. So when you create a dataset with it the "bottom left" pixel is always [1, 1]. Thus we center the crpix such that it's correctly within this newly shifted cutout and at the location of the DiaSource in the image. There is a possible future feature in astropy to add an origin to the CCDData object but it is currently not a priority.

Reply:
I think "not a priority" is right for them but if we want to add that feature they seem like they would be more than happy to accept a patch. We have been in discussion for years to add pixel origin to NDData.